### ISSUE TITLE
unconnected peers hold on to nicknames indefinitely

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#2057]
```
host1:~$ weave launch && weave consense
host2:~$ weave launch 192.168.48.11 && weave consense
host2:~$ weave reset
host1:~$ weave rmpeer host2
No entries for peer found
host1:~$ weave rmpeer host3
Cannot find peer 'host3'
```
The difference in the error message indicates that host2 was still known to host1. The reason is that we only prune nicknames from ipam on receipt of gossip. When a peer is no longer connected to any others, it does not receive any gossip, so any nicknames left at the time will remain forever.

This is of little practical consequence, other than the possibly surprising difference in error message.

However, it's also easy to fix by pruning on a PeerGone event from the mesh.

**Comment by brb** [comment]
Which version are you running? From `weave consense` I can spot that it is not the latest one.

With the latest master branch (2f5a3f4513d5f2578ea74a38ebace94484a7ab11) I cannot reproduce the error messages. Mainly, because the current implementation of `AdminTakeoverRanges` returns a count of transferred IP addrs instead of error, if a peer does not exist.

```bash
host1:~$ weave launch && weave prime
host2:~$ weave launch 192.168.48.11 && weave prime
host2:~$ weave reset
host1:~$ weave rmpeer host2
0 IPs taken over from host2
host1:~$ weave rmpeer host5
0 IPs taken over from host5
```

Also, please note that the first error message from your run is no longer in use:
```
$ ack --ignore-dir .pkg 'No entries for peer found'
ipam/ring/ring.go
43:     ErrNotFound        = errors.New("No entries for peer found")
$ ack --ignore-dir={.pkg,vendor} 'ErrNotFound'
ipam/ring/ring.go
43:     ErrNotFound        = errors.New("No entries for peer found")
```

**Comment by rade** [comment]
We changed the rmpeer behaviour, and hence the issue is no longer visible to the user. But it is still there.

**Comment by rade** [comment]
> note that the first error message from your run is no longer in use:

Well spotted!

Fixed in 9326775.
